libs/langchain/langchain/schema/agent.py

```diff
@@ -16,9 +27,23 @@ class AgentAction:
     """Additional information to log about the action."""
 
 
-class AgentFinish(NamedTuple):
+class AgentActionMessageLog(AgentAction):
+    message_log: List[BaseMessage]
```

Comment:
This feels a bit hacky in terms of ontology. It's not obvious why an AgentAction should know about messages. 

It might help to add information saying that this is a version of the action that can be used for chat models to help pass around information about the context of the action.

```suggestion
    messages: Sequence[BaseMessage]
```